### PR TITLE
Task-58887: Delete eXo event in Personal exchange agenda 

### PR DESCRIPTION
**Pull request by azayati**

### CODE REVIEW

File: agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java [hunk 1]
```diff
@@ -74,4 +78,8 @@ public void deleteExchangeSetting(long userIdentityId) {
                                ExchangeConnectorUtils.EXCHANGE_PASSWORD_KEY);
   }
 
+  public  void deleteRemoteEvent(long eventId, long userIdentityId){
```
Comment (azayati) [inline agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java:81]:
```suggestion
  public void deleteRemoteEvent(long eventId, long userIdentityId){
```

File: agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/service/ExchangeConnectorServiceImpl.java [hunk 2]
```diff
@@ -211,33 +196,32 @@ public EventEntity getRemoteEventById(long eventId, long identityId, ZoneId user
...
-      ExchangeUserSetting exchangeUserSetting = getExchangeSetting(eventAttendee.getIdentityId());
+  public void deleteExchangeEvent(long userIdentityId, long eventId) throws IllegalAccessException {
+      RemoteEvent remoteEvent = agendaRemoteEventService.findRemoteEvent(eventId, userIdentityId);
+      ExchangeUserSetting exchangeUserSetting = getExchangeSetting(userIdentityId);
       try (ExchangeService exchangeService = new ExchangeService(ExchangeVersion.Exchange2010_SP2)) {
         connectExchangeServer(exchangeService, exchangeUserSetting);
         ItemId itemId = new ItemId(remoteEvent.getRemoteId());
         Appointment appointment = Appointment.bind(exchangeService, itemId);
         appointment.delete(DeleteMode.MoveToDeletedItems);
+        exchangeConnectorStorage.deleteRemoteEvent( eventId, userIdentityId);
```
Comment (azayati) [inline agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/service/ExchangeConnectorServiceImpl.java]:
```suggestion
        exchangeConnectorStorage.deleteRemoteEvent(eventId, userIdentityId);
```

File: agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java [hunk 3]
```diff
@@ -26,8 +27,11 @@ public class ExchangeConnectorStorage {
 
   private SettingService      settingService;
 
-  public ExchangeConnectorStorage(SettingService settingService) {
+  private AgendaRemoteEventStorage remoteEventStorage;
```
Comment (azayati) [inline agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java:30]:
```suggestion
  private AgendaRemoteEventStorage agendaRemoteEventStorage;
```

File: agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java [hunk 4]
```diff
@@ -26,8 +27,11 @@ public class ExchangeConnectorStorage {
...
+
+  public ExchangeConnectorStorage(SettingService settingService, AgendaRemoteEventStorage remoteEventStorage) {
```
Comment (azayati) [inline agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/storage/ExchangeConnectorStorage.java:32]:
```suggestion
  public ExchangeConnectorStorage(SettingService settingService, AgendaRemoteEventStorage agendaRemoteEventStorage) {
```

**Review by azayati — COMMENTED** [review]
See review

**Review by IlhemEssaadi — APPROVED** [review]

**Pull request merged**